api/v1/controllers/ProjectController.js

```diff
@@ -0,0 +1,165 @@
+var _ = require('lodash');
+var bodyParser = require('body-parser');
+var middleware = require('../middleware');
+var router = require('express').Router();
+
+var errors = require('../errors');
+var config = require('../../config');
+var requests = require('../requests');
+var roles = require('../utils/roles');
+
+var ProjectService = require('../services/ProjectService');
+var PermissionService = require('../services/PermissionService');
+
+
+function _validGetAllRequest(page, count, published){
+	if(_.isNaN(page)){
+		var message = "Invalid page parameter";
+		var source = "page";
+		throw new errors.InvalidParameterError(message, source);
+	}
+	if(_.isNaN(count)){
+		var message = "Invalid count parameter";
+		var source = "count";
+		throw new errors.InvalidParameterError(message, source);
+	}
+	if(_.isNaN(published) || (published != 0 && published != 1)){
+		var message = "Invalid published parameter";
+		var source = "published";
+		throw new errors.InvalidParameterError(message, source);
+	}
+}
+
+function createProject (req, res, next) {
+	attributes = req.body;
+
+	PermissionService
+		.canCreateProject(req.user)
+		.then(function (isAuthed) {
+			return ProjectService.createProject(attributes);
+		})
+		.then(function (newProject) {
+			res.body = newProject.toJSON();
+
+			next();
+			return null;
+		})
+		.catch(function (error){
+			next(error);
+			return null;
+		});
+}
+
+function getProject (req, res, next) {
+	var id = req.params.id;
+
+	ProjectService
+		.findProjectById(id)
+		.then(function (project) {
+			res.body = project.toJSON();
+
+			next();
+			return null;
+		})
+		.catch(function (error) {
+			next(error);
+			return null;
+		});
+}
+
+function getAllProjects (req, res, next) {
+	var page = parseInt(req.params.page);
+	var count = parseInt(req.query.count);
+	var published = parseInt(req.query.published);
```

Comment:
What does these default to when the parameter is not provided, `NaN`? If so, perhaps we should provide defaults instead of throwing errors. For example, if I don't provide the `published` parameter, it seems reasonable to expect that this would at least default to true. I believe `var published = parseInt(req.query.published) || 1` would do the trick.

That'd allow you to get rid of the parameter checking done in the service.